$NetBSD: patch-configure.ac,v 1.2 2022/12/12 22:07:05 sekiya Exp $

* Add support for blocklist(blacklist).
* Remove WRKDIR reference from config.h.

--- configure.ac.orig	2022-11-07 22:17:42.020258069 +0000
+++ configure.ac
@@ -36,7 +36,9 @@ AC_DEFINE([PACKAGE_DESCRIPTION], [m4_ifn
 AC_DEFINE([PACKAGE_SRCID], ["][bind_SRCID]["], [A short hash from git])
 
 bind_CONFIGARGS="${ac_configure_args:-default}"
-AC_DEFINE_UNQUOTED([PACKAGE_CONFIGARGS], ["$bind_CONFIGARGS"], [Either 'defaults' or used ./configure options])
+faked_CONFIGARGS=$(echo ${bind_CONFIGARGS} | \
+	sed -e 's|@TOOLS_DIR@|@PREFIX@|g' -e 's|@BUILDLINK_DIR@|@PREFIX@|g')
+AC_DEFINE_UNQUOTED([PACKAGE_CONFIGARGS], ["$faked_CONFIGARGS"], [Either 'defaults' or used ./configure options])
 
 AC_DEFINE([PACKAGE_BUILDER], ["make"], [make or Visual Studio])
 
@@ -540,6 +542,16 @@ AC_CHECK_FUNCS([sched_yield pthread_yiel
 AC_CHECK_FUNCS([pthread_setname_np pthread_set_name_np])
 AC_CHECK_HEADERS([pthread_np.h], [], [], [#include <pthread.h>])
 
+# libblocklist(libblacklist)
+AC_MSG_CHECKING(for libblocklist)
+AC_CHECK_HEADER([blocklist.h],
+  [AC_DEFINE([HAVE_BLOCKLIST_H], [1], [blocklist support])
+   AC_CHECK_LIB(blocklist, blocklist)],
+  [AC_MSG_CHECKING(for libblacklist)
+   AC_CHECK_HEADER([blacklist.h],
+   [AC_DEFINE([HAVE_BLACKLIST_H], [1], [blacklist support])
+    AC_CHECK_LIB(blacklist, blacklist)])])
+
 # libuv
 AC_MSG_CHECKING([for libuv])
 PKG_CHECK_MODULES([LIBUV], [libuv >= 1.0.0], [],
@@ -1212,6 +1224,18 @@ AS_IF([test "$enable_fixed_rrset" = "yes
       [AC_DEFINE([DNS_RDATASET_FIXED], [1],
 		 [Define to enable "rrset-order fixed" syntax.])])
 
+# blocklist
+AC_ARG_WITH([blocklist],
+	    [AS_HELP_STRING([--with-blocklist],
+		[enable blocklist support [yes|no|detect] (default is detect)])],
+		[], [with_blocklist="$with_blocklist"])
+
+# blacklist
+AC_ARG_WITH([blacklist],
+	    [AS_HELP_STRING([--with-blacklist],
+		[enable blacklist support [yes|no|detect] (default is detect)])],
+		[], [with_blacklist="$with_blacklist"])
+
 #
 # Activate dnstap?
 #
